File: mobile/app/src/main/java/com/bytecats/metanoia/models/BibleModels.kt
package com.bytecats.metanoia.models

data class BibleBook(val name: String, val chapters: Int, val testament: String, val isApocrypha: Boolean = false)
data class InterlinearWord(val original: String, val strongs: String, val translation: String)
data class Favorite(val strongs: String, val lemma: String, val definition: String)
data class Highlight(val book: String, val chapter: Int, val verse: Int, val color: Int)
data class Note(val id: Int = 0, val book: String, val chapter: Int, val verse: Int, val content: String, val timestamp: Long)
data class SearchResult(val book: String, val chapter: Int, val verse: Int, val text: String)

data class Verse(val number: Int, val text: String)
data class LexiconEntry(val lemma: String, val definition: String)

data class LibraryStats(
    val versesOt: Int, val versesNt: Int,
    val lexiconHeb: Int, val lexiconGk: Int,
    val notesCount: Int, val highlightsCount: Int,
    val interlinearCount: Int,
    val dbSizeMb: Double
)

File: ios/Metanoia/Metanoia/Models/BibleModels.swift
import Foundation

struct Verse: Identifiable, Codable, Hashable {
    let book: String
    let chapter: Int
    let number: Int
    let text: String
    var id: String { "\(book)-\(chapter)-\(number)" }
    var reference: String { "\(book) \(chapter):\(number)" }
}

struct InterlinearWord: Identifiable, Codable {
    let book: String
    let chapter: Int
    let verse: Int
    let wordIndex: Int
    let originalText: String
    let translation: String
    let strongs: String
    var id: String { "\(book)-\(chapter)-\(verse)-\(wordIndex)" }
}

struct Favorite: Identifiable, Codable {
    let strongs: String
    let lemma: String
    let definition: String
    var id: String { strongs }
}

struct Highlight: Identifiable, Codable {
    let book: String
    let chapter: Int
    let verse: Int
    let color: Int
    var id: String { "\(book)-\(chapter)-\(verse)" }
}

struct Note: Identifiable, Codable {
    let id: Int64
    let book: String
    let chapter: Int
    let verse: Int
    let content: String
    let timestamp: Date
}

struct SearchResult: Identifiable {
    let book: String
    let chapter: Int
    let verse: Int
    let text: String
    var id: String { "\(book)-\(chapter)-\(verse)" }
}

struct ReadingProgress: Codable {
    let book: String
    let chapter: Int
    let firstReadAt: Date?
    let lastReadAt: Date?
    let readCount: Int
    let readingTimeSeconds: Int
}

struct LibraryStats {
    let versesOt: Int
    let versesNt: Int
    let lexiconHeb: Int
    let lexiconGk: Int
    let notesCount: Int
    let highlightsCount: Int
    let interlinearCount: Int
    let dbSizeMb: Double
}

struct HotChapter: Identifiable {
    let book: String
    let chapter: Int
    let views: Int
    var id: String { "\(book)-\(chapter)" }
}
